### PR TITLE
chore: Update datafusion

### PR DESCRIPTION
**Pull request by alamb**
# Rationale:
1. Reduce pruning overhead via https://github.com/apache/arrow-datafusion/pull/1700 and https://github.com/apache/arrow-datafusion/pull/1695

# Changes:
1. Update to https://github.com/apache/arrow-datafusion/commit/f849968057ddddccc9aa19915ef3ea56bf14d80d
2. Update to use new DataFusion API
2. Remove an unnecessary creation of `ExecutionContextState` (from https://github.com/influxdata/influxdb_iox/pull/3579)

### CODE REVIEW

File: query/src/exec/split.rs [hunk 1]
```diff
@@ -605,27 +607,4 @@ mod tests {
             MemoryExec::try_new(&partitions, schema, projection).expect("Created MemoryExec");
         Arc::new(input)
     }
-
-    fn compile_expr(input: &dyn ExecutionPlan, expr: Expr) -> Arc<dyn PhysicalExpr> {
-        let physical_planner = DefaultPhysicalPlanner::default();
-        // we should probably get a handle to the one in the default
```
Comment (alamb) [inline query/src/exec/split.rs:611]: The day has arrived!

File: query/src/util.rs [hunk 2]
```diff
@@ -68,11 +69,7 @@ pub fn df_physical_expr(
     input: &dyn ExecutionPlan,
     expr: Expr,
 ) -> std::result::Result<Arc<dyn PhysicalExpr>, DataFusionError> {
-    // To create a physical expression for a logical expression we need appropriate
-    // PhysicalPlanner and ExecutionContextState, however, our given logical expression is very basic
-    // and any planner or context will work
-    let physical_planner = DefaultPhysicalPlanner::default();
-    let ctx_state = datafusion::execution::context::ExecutionContextState::new();
+    let execution_props = ExecutionProps::new();
```
Comment (alamb) [inline query/src/util.rs:72]: ✅

File: datafusion_util/src/lib.rs [hunk 3]
```diff
@@ -211,16 +211,16 @@ pub fn stream_from_batch(batch: RecordBatch) -> SendableRecordBatchStream {
...
+    let stream = SizedRecordBatchStream::new(batches[0].schema(), batches, mem_metrics);
     Box::pin(stream)
 }
 
 /// Create a SendableRecordBatchStream that sends back no RecordBatches with a specific schema
 pub fn stream_from_schema(schema: SchemaRef) -> SendableRecordBatchStream {
     let dummy_metrics = ExecutionPlanMetricsSet::new();
-    let baseline_metrics = BaselineMetrics::new(&dummy_metrics, 0);
-    let stream = SizedRecordBatchStream::new(schema, vec![], baseline_metrics);
+    let mem_metrics = MemTrackingMetrics::new(&dummy_metrics, 0);
```
Comment (alamb) [inline datafusion_util/src/lib.rs:222]: Required by https://github.com/apache/arrow-datafusion/pull/1691

**Review by carols10cents — APPROVED** [review]
LGTM!

**Pull request merged**